frontend/templates/troubleshooting/index.tsx

```diff
@@ -128,7 +134,7 @@ function TableOfContents({ solutions }: { solutions: Section[] }) {
          {solutions.map((solution, index) => (
             <Link
                key={solution.heading}
-               href={`#solution-${index + 1}`}
+               href={`#${solution.heading.toLowerCase().replace(/\s/g, '-')}`}
```

Comment:
I think we should extract this into a utils function: `solutionHeadingToId(solution.heading)` so we can be sure the same format is used in both places.

Reply:
Looks like the value [ought to be url-encoded](https://stackoverflow.com/questions/20679500/how-to-encode-space-in-the-fragment-identifier-in-a-url) as well (to handle things besides spaces).

Reply:
https://github.com/iFixit/react-commerce/pull/1707/commits/b7107507ae64bd718acc49fe6085bfdb3d5eb4ee